src/Simplex/Chat/Remote.hs

```diff
@@ -342,19 +341,58 @@ liftRH rhId = liftError (ChatErrorRemoteHost (RHId rhId) . RHEProtocolError)
 
 -- * Mobile side
 
+-- ** Multicast
+
 findKnownRemoteCtrl :: ChatMonad m => m ()
-findKnownRemoteCtrl = undefined -- do
+findKnownRemoteCtrl = handleCtrlError "findKnownRemoteCtrl" $ do
+  knownCtrls <- withStore' getRemoteCtrls
+  pairings <- maybe (throwError $ ChatErrorRemoteCtrl RCENoKnownControllers) (pure . fmap (\RemoteCtrl {ctrlPairing} -> ctrlPairing)) $ nonEmpty knownCtrls
+  withRemoteCtrlSession_ $ maybe (Right ((), Just RCSessionStarting)) (\_ -> Left $ ChatErrorRemoteCtrl RCEBusy)
+  foundCtrl <- newEmptyTMVarIO
+  cmdOk <- newEmptyTMVarIO
+  action <- async $ handleCtrlError "findKnownRemoteCtrl.discover" $ do
+    atomically $ takeTMVar cmdOk
+    (RCCtrlPairing {ctrlFingerprint}, inv) <- timeoutThrow (ChatErrorRemoteCtrl RCETimeout) (networkIOTimeout * 4) . withAgent $ \a -> rcDiscoverCtrl a pairings
+    rc_ <- withStore' $ \db -> getRemoteCtrlByFingerprint db ctrlFingerprint
+    rc <- maybe (throwChatError $ CEInternalError "connecting with a stored ctrl") pure rc_
```

Comment:
this maybe (throw) is hard to read... I thin it's better just use \case in this case, without moving Maybe to ExceptT territory, they are different kind of error.

```suggestion
    withStore' (`getRemoteCtrlByFingerprint` ctrlFingerprint) >>= \case
      Nothing -> throwChatError $ CEInternalError "connecting with a stored ctrl"
      Just rc -> do
```

same as above